GEODE-7949: Geode Redis - Get/Set commands for RedisString datatype to cover new parameters

xtreme-prasath opened the pull request:
Get/Set commands for RedisString datatype to cover new parameters

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

sabbeyPivotal commented on geode-redis/src/integrationTest/java/org/apache/geode/redis/StringsIntegrationTest.java:
```diff
@@ -92,13 +92,282 @@ public static void tearDown() {
+    String result_EX = jedis.set(key, value, setParamsEX);
+    assertThat(result_EX).isEqualTo("OK");
+    assertThat(jedis.ttl(key)).isGreaterThan(15L);
+    jedis.set(key, value);
```
Setting a key here seems unnecessary since the setting with XX  on line 215 should clear TTL.

sabbeyPivotal commented on geode-redis/src/integrationTest/java/org/apache/geode/redis/StringsIntegrationTest.java:
```diff
@@ -92,13 +92,282 @@ public static void tearDown() {
+    SetParams setParamsEX = new SetParams();
+    setParamsEX.ex(secondsUntilExpiration);
+
+    SetParams setParamsXX = new SetParams();
+    setParamsXX.xx();
```
This parameter is not used.

xtreme-prasath commented:
added tests for negative EX/PX parameter 
added integration tests for KEEPTTL however Jedis doesn't support KEEPTTL yet, so I have that test ignored for now. 
added unit tests for GET executor for too few arguments and too many arguments

prettyClouds commented on geode-redis/src/integrationTest/java/org/apache/geode/redis/StringsIntegrationTest.java:
```diff
@@ -92,13 +93,328 @@ public static void tearDown() {
+  }
+
+  @Test
+  public void testSET_shouldAccept_NX_argumentToOnlySetKeyIfIt_Does_not_Exist() {
```
the names of these tests are inconsistent:
here Does_not_Exist has underscores when the convention suggests no underscores and camelCase
the word HONOR used above seems inconsistent with the other language of the the test names.

xtreme-prasath replied:
done

prettyClouds commented on geode-redis/src/integrationTest/java/org/apache/geode/redis/StringsIntegrationTest.java:
```diff
@@ -270,10 +586,10 @@ public void testGetSet_whenWrongType_shouldReturnError() {
 
   @Test
   public void testSet_keyExistsWithDifferentDataType_returnsRedisDataTypeMismatchException() {
```
i think this test belongs in the HSET tests, not in the strings.  
Consider moving it?
Another option might be moving all the tests that re-write keys into a single test, so that context is in one place.

prettyClouds replied:
also, we might want to test that set overrides the hset, if we are not doing so already

xtreme-prasath replied:
done

dschneider-pivotal commented on geode-redis/src/main/java/org/apache/geode/redis/internal/KeyRegistrar.java:
```diff
@@ -76,16 +76,26 @@ public void validate(ByteArrayWrapper key, RedisDataType type) {
+   * @param key Key to check
+   */
+  public boolean isProtected(ByteArrayWrapper key) {
+    return redisMetaRegion.containsKey(key.toString())
```
I think the extra containsKey call is not needed in this get. Region.get will return null if a region does not contain the key and null will not be equal to REDIS_PROTECTED.
So I advise you get rid of the containsKey call.

xtreme-prasath replied:
Good call Darrel. Addressed that.
